locksmith/src/app.js

```diff
@@ -0,0 +1,66 @@
+const bodyParser = require('body-parser')
+const cors = require('cors')
+const express = require('express')
+const tokenMiddleware = require('./token_middleware')
+const Lock = require('./lock')
+
+const app = express()
+const router = express.Router()
+
+router.put('/lock/:lockAddress', function(req, res) {
+  let newAddress = req.body.address
+  let tempAddress = req.params.lockAddress
+
+  if (tempAddress && newAddress) {
+    Lock.findOne({
+      where: { address: tempAddress, owner: req.owner },
+      raw: true,
+    })
+      .then(result => {
+        result.address = newAddress
```

Comment:
@julien51 I believe in a scenario like this the potential attacker would receive a 412 back. Our validation middleware injects the signee/owner to the express application. If the message signee is not the owner of the object they are attempting to modify then the lookup will return an empty result

Reply:
Ok, I did not see a test for this, maybe I am missing it?
Also I still think it would be a quick check here too and fail if the lock we're moving to exists? Better safe than sorry ;p

Reply:
All good, now addressed in coverage